### ISSUE TITLE
Allow creation and mutation of elements, attrs etc

### ISSUE THREAD

**Issue by hadley** [hadley/xml2#9]
If you need this, please :+1: this issue with a brief use case

**Comment by jread-usgs** [comment]
:+1: I would use this to 1) build xml for POSTs 2) modify svg elements or programmatically build interactive svgs.

**Comment by jennybc** [comment]
:+1: In [`gspreadr`](https://github.com/jennybc/gspreadr), which wraps the Google Sheets API, we have a function to edit cells in the sheet. The [workflow is this](https://github.com/jennybc/gspreadr/blob/f5eb3394ca632608be0bc3f0c5acdef38163a857/R/edit-data.R#L73-L96): fetch the current cell contents, then edit specific values in the XML, send back via POST. Right now we're using the XML package but I'd rather be using this package.

**Comment by jennybc** [comment]
My first use case was of creating XML sort of *de novo*. I just found another one that's more of a true mutate. And [our current workflow](https://github.com/jennybc/gspreadr/blob/a0f68769260a96374b32d0a565811a5b84935a42/R/edit-spreadsheets.R#L386-L414) is to just use regex substitution :flushed: If this were easier in one of the XML packages, like yours, I'd much prefer to do that.

**Comment by timelyportfolio** [comment]
:+1: I do this all the time especially with `SVG` from `gridSVG` and `dev.svg` where I might change `height`, `width`, `viewBox`, `style` attributes.

**Comment by sjp** [comment]
To add to what @timelyportfolio mentioned, it would be nice to be able to build up an image in `gridSVG` using something other than `XML`. I suspect that `xml2` could greatly improve the performance of SVG image generation.

Additionally, I've been making changes to `selectr` so that it has plugin packages for `XML` and `xml2`. Supporting creation/mutation would be useful because the plugin packages might be good for manipulating scraped data. The same applies for `rvest`.

**Comment by nicolasCoutin** [comment]
:+1:  

- I've done all sorts of crazyness playing with R-bindings of [Snap.svg](http://snapsvg.io/). when it doesn't explode it's very powerful!

**Comment by nicolasCoutin** [comment]
Say you want to take [this vignette](http://cran.r-project.org/web/packages/dplyr/vignettes/data_frames.html) and modify/enhance it so that when you hover over each function it displays the function's Arguments documentation (so that if past you has been less explicit than present you would like you can make sense of what's what without having to open up your IDE).

- functions are wrapped with a `<span>` of class `kw`
- create a related `<table>` element that contains the Arguments and their explanation
- default this element with a style such that it is hidden
- add `:hover` style behaviour to reveal!

if you could...

```r
h_con <- 'http://cran.r-project.org/web/packages/dplyr/vignettes/data_frames.html'
modified_html <- h_con %>% read_html() %>%  ...
```

where `...` is some combination of ?

```r
node_select() # == rvest::html_node() 
nodes_select() # == rvest::html_nodes() 
node_mutate()
nodes_mutate()
nodes_create()
... and friends?
```

**Comment by cdiener** [comment]
:+1:

We use xml2 to handle biochemical reaction models for Systems Biology described in SBML (http://sbml.org). Writing those models to disk is kind of crucial for reproducible research so it would be nice feature avoid writing all the tags etc. by hand.

**Comment by renkun-ken** [comment]
My use case is to batch create a group of xml configurations to deploy to server programs. The xml contains some nodes with simple structure just like typical configuration files.

**Comment by cboettig** [comment]
:+1: I'm working on several packages designed to assist researchers both consume and publish datasets with XML-based metadata standards (e.g. EML, NeXML).  The basic idea is that it's much easier to discover interesting datasets when you can search a big pool of them over interesting metadata fields.  The data repository managers have made fancy tools to do the search and discovery, but pretty much no one wants to write up their metadata in XML.  Such R packages would let users turn nice familiar R structures into the required XML easily; complementing the current use case of `xml2` as a way to turn XML data into nice R structures.  This helps researchers share & organize their own data according to existing standards, rather than just be consumers.

**Comment by jennybc** [comment]
Just recording something already conveyed via email ...

An `xml2` option for XML creation and mutation would be a welcome improvement over `XML`, specifically in terms of better handling of encoding.

And, for that matter, hopefully `xml2` would also make it easier to finesse namespaces when creating XML (see e.g. [this SO thread](http://stackoverflow.com/questions/32939229/creating-xml-in-r-with-namespaces/32941524#32941524)).

**Comment by imanuelcostigan** [comment]
Use case for me: reading in XML file containing market data, modifying values and attributes to represent "simulated" market data and write modified XML back out for use by another system

**Comment by jimhester** [comment]
Closed by #76
